sktime/transformations/panel/catch22.py

```diff
@@ -0,0 +1,1126 @@
+        for i in range(1, len(X)):
+            cs[i] = cs[i - 1] + X[i]
+
+        return fluct_prop(cs, len(X), False)
+
+    @staticmethod
+    @njit(fastmath=True, cache=True)
+    def SB_TransitionMatrix_3ac_sumdiagcov(X, acfz):
+        # Trace of covariance of transition matrix between symbols in 3-letter alphabet
+        ds = np.zeros(int((len(X) - 1) / acfz + 1))
+        for i in range(len(ds)):
+            ds[i] = X[i * acfz]
+        indicies = np.argsort(ds)
+
+        bins = np.zeros(len(ds), dtype=np.int32)
+        q1 = int(len(ds) / 3)
+        q2 = q1 * 2
+        for i in range(q1 + 1, q2 + 1):
+            bins[indicies[i]] = 1
+        for i in range(q2 + 1, len(indicies)):
+            bins[indicies[i]] = 2
+
+        t = np.zeros((3, 3))
+        for i in range(len(ds) - 1):
+            t[bins[i + 1]][bins[i]] += 1
+        t /= len(ds) - 1
+
+        means = np.zeros(3)
+        for i in range(3):
+            means[i] = np.mean(t[i])
+
+        cov = np.zeros((3, 3))
+        for i in range(3):
+            for n in range(3):
+                covariance = 0
+                for j in range(3):
+                    covariance += (t[i][j] - means[i]) * (t[n][j] - means[n])
+                covariance /= 2
+
+                cov[i][n] = covariance
+                cov[n][i] = covariance
+
+        ssum = 0
+        for i in range(3):
+            ssum += cov[i][i]
+
+        return ssum
+
+    @staticmethod
+    @njit(fastmath=True, cache=True)
+    def PD_PeriodicityWang_th0_01(X):
+        # Periodicity measure of (Wang et al. 2007)
+        y_spline = spline_fit(X)
+
+        y_sub = np.zeros(len(X))
+        for i in range(len(X)):
+            y_sub[i] = X[i] - y_spline[i]
+
+        acmax = int(np.ceil(len(X) / 3.0))
+        acf = np.zeros(acmax)
+        for tau in range(1, acmax + 1):
+            covariance = 0
+            for i in range(len(X) - tau):
+                covariance += y_sub[i] * y_sub[i + tau]
+            acf[tau - 1] = covariance / (len(X) - tau)
+
+        troughs = np.zeros(acmax, dtype=np.int32)
+        peaks = np.zeros(acmax, dtype=np.int32)
+        n_troughs = 0
+        n_peaks = 0
+        for i in range(1, acmax - 1):
+            slope_in = acf[i] - acf[i - 1]
+            slope_out = acf[i + 1] - acf[i]
+
+            if slope_in < 0 and slope_out > 0:
+                troughs[n_troughs] = i
+                n_troughs += 1
+            elif slope_in > 0 and slope_out < 0:
+                peaks[n_peaks] = i
+                n_peaks += 1
+
+        out = 0
+        for i in range(n_peaks):
+            j = -1
+            while troughs[j + 1] < peaks[i] and j + 1 < n_troughs:
+                j += 1
+
+            if j == -1 or acf[peaks[i]] - acf[troughs[j]] < 0.01 or acf[peaks[i]] < 0:
+                continue
+
+            out = peaks[i]
+            break
+
+        return out
+
+
+@njit(fastmath=True, cache=True)
+def histogram_mode(X, num_bins, smin, smax):
```

Comment:
for the functions below, I suggest to make them private (leading underscore)

Reply:
besides that I think this is good to go

Reply:
Done